what4/src/What4/Expr/Builder.hs

```diff
@@ -3686,26 +3686,29 @@ semiRingAdd sym sr x y =
 
       (SR_Constant xc, _)
         | Just (BaseIte _ _ cond a b) <- asApp y
-        -> do xa <- semiRingAdd sym sr x a
-              xb <- semiRingAdd sym sr x b
-              semiRingIte sym sr cond xa xb
-
-        | otherwise
-        -> sum' sym (WSum.addConstant sr (WSum.var sr y) xc)
+        , isConstantSemiRingExpr a || isConstantSemiRingExpr b -> do
```

Comment:
@robdockins This might be a good target for `hlint` or a shell script running in Travis. It's definitely something I've added in PRs, as I didn't realize there was a preference against it.

Reply:
I have a preference for it, so this is new to me as well.
Maybe there should be a style guide for crucible?  There seems to be a style war within this file (not just this but also position of `=>` and `->` in declarations).